ui/packages/catalog-ui-search/src/main/webapp/js/model/List.js

```diff
@@ -53,13 +54,56 @@ function generateCql(bookmarks) {
   })
 }
 
+function simplifyListFilter(listFilters) {
+  if (!listFilters) {
+    return listFilters
+  }
+  let filtersArray = listFilters.filters || [listFilters]
+  filtersArray = filtersArray
+    .filter(filter => !filter.filters) //Only supports one filter group type, no nested filter groups
+    .map(filter => {
+      return {
+        ...filter,
+        value:
+          filter.value && filter.value.value
+            ? filter.value.value
+            : filter.value,
+      }
+    })
+  return {
+    type: listFilters.filters ? listFilters.type : 'AND',
+    filters: filtersArray,
+  }
+}
+
+function parseList(data) {
+  // for backwards compatability
+  if (data['list.cql']) {
```

Comment:
✏️ we can move the check on L83 here to reduce the amount of nesting
```suggestion
  if (data['list.cql'] && !data['list.filters']) {
```

Reply:
Had it there originally, but since no matter what, we always want to delete list.cql, hence why moved it bellow. Else if scenario of if for some reason there is a list.cql and a list.filters, we will delete that list.cql.